indexd/index/drivers/alchemy.py

```diff
@@ -371,6 +375,88 @@ def ids(self,
 
             return [i.to_document_dict() for i in query]
 
+    @staticmethod
+    def _negate_filter(session,
+                       query,
+                       urls=None,
+                       acl=None,
+                       file_name=None,
+                       version=None,
+                       metadata=None,
+                       urls_metadata=None):
+        """
+        param_values passed in here will be negated
+
+        for string (version, file_name), filter with value != <value>
+        for list (urls, acl), filter with doc that don't HAS <value>
+        for dict (metadata, urls_metadata). In each (key,value) pair:
+        - if (not value is True): then filter with key doesn't exist
+        - if value is provided, then filter with value != <value> OR key doesn't exist
+
+        :param session: db session
```

Comment:
we use [google docstring style](https://github.com/uc-cdis/rfcs/blob/master/text/0003-style-guide.md#documentation)